### PR TITLE
replace moralis balances query with covalent

### LINKED ISSUE #2947: [Tech Debt]: Replace Moralis Query for Token Balances
### What is needed?

need to replace the getWalletTokenBalances() moralis call with a call that does not depend on moralis

### PR DESCRIPTION
**Pull request by benwolski**
### Describe your changes 
replace moralis balances query with covalent

### Link the related issue
Closes #2947

### Checklist before requesting a review
- [x] Is this PR ready for merge? (Please convert to a draft PR otherwise)
- [x] I have performed a self-review of my code.
- [x] Did I request feedback from a team member prior to merge? 
- [x] Does my code following the style guide at `docs/CODING-STYLE.md`?

### Instructions for Reviewer
**Functionalities or workflows that should specifically be tested.**

1. balances in wallet dropdown, token selector and portfolio should update after any transaction

2.

**Environmental conditions which may result in expected but differential behavior.**

1.

2.

### If relevant, list additional work to complete pre-merge (delete logging, code abstraction, etc)

### CODE REVIEW

File: src/App/functions/fetchTokenBalances.ts [hunk 1]
```diff
@@ -133,38 +144,39 @@ export const fetchErc20TokenBalances = async (
...
-    ): TokenIF => {
-        const moralisErc20Balance = BigNumber.from(
-            erc20value.amount.toString(),
-        );
-        const moralisErc20BalanceDisplay = toDisplayQty(
-            moralisErc20Balance,
-            erc20value.decimals,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const getTokenInfoFromCovalentBalance = (tokenBalance: any): TokenIF => {
+        console.log({ tokenBalance });
```
Comment (enwong10) [inline src/App/functions/fetchTokenBalances.ts:149]: let's remove the console.log

File: src/components/Global/Account/AccountTabs/Exchange/Exchange.tsx [hunk 2]
```diff
@@ -29,12 +29,8 @@ export default function Exchange(props: propsIF) {
...
     const { tokens } = useContext(TokenContext);
 
-    const { nativeToken, erc20Tokens } = useAppSelector(
-        (state) => state.userData.tokens,
-    );
-    const connectedUserTokens = [nativeToken]
-        .concat(erc20Tokens)
-        .filter((token) => token);
+    const { erc20Tokens } = useAppSelector((state) => state.userData.tokens);
+    const connectedUserTokens = (erc20Tokens ?? []).filter((token) => token);
```
Comment (enwong10) [inline src/components/Global/Account/AccountTabs/Exchange/Exchange.tsx:33]: what exactly is this filter doing?

File: src/components/Global/Account/AccountTabs/Wallet/Wallet.tsx [hunk 3]
```diff
@@ -29,12 +29,10 @@ export default function Wallet(props: propsIF) {
...
-    const { nativeToken, erc20Tokens } = useAppSelector(
-        (state) => state.userData.tokens,
-    );
-    const connectedUserTokens: Array<TokenIF | undefined> = [nativeToken]
-        .concat(erc20Tokens)
-        .filter((token) => token);
+    const { erc20Tokens } = useAppSelector((state) => state.userData.tokens);
+    const connectedUserTokens: Array<TokenIF | undefined> = (
+        erc20Tokens ?? []
+    ).filter((token) => token);
```
Comment (enwong10) [inline src/components/Global/Account/AccountTabs/Wallet/Wallet.tsx:35]: 👀

File: src/components/Global/Account/AccountTabs/Exchange/Exchange.tsx (hunk 2 above)
Comment (benwolski) [reply src/components/Global/Account/AccountTabs/Exchange/Exchange.tsx]: no idea...removed

File: src/components/Global/Account/AccountTabs/Wallet/Wallet.tsx (hunk 3 above)
Comment (benwolski) [reply src/components/Global/Account/AccountTabs/Wallet/Wallet.tsx]: also removed

**Review by enwong10 — APPROVED** [review]
Code LGTM! 🚀

**Review by enwong10 — APPROVED** [review]
🚀

**Pull request merged**